### DISCUSSION TITLE
Frigate in LXC container without Docker in lxc nor Proxmox

### DISCUSSION THREAD

**Discussion by GnomiBerlin** [blakeblackshear/frigate#13736] [Ask A Question]
Hello,
When reading the LXC hints for Frigate, it seems that always a docker setup within the LXC container is meant?!
Is there no blank LXC container version without this docker overhead and restrictions possible?

**Comment by NickM-27** [comment]
the tteck script can be used to install without docker, but any form of using proxmox is not officially supported. There are some differences in the way it behaves and reported issues with the proxmox kernel.

**Reply by GnomiBerlin** [reply]
Thank you, I wanted to use Frigate neither with Docker nor Proxmox - just LXC. 
I will check the mentioned script how to do it

**Reply by GnomiBerlin** [reply]
so, checked the @tteck description and there is in the installation hints:

apt install docker.io
systemctl enable docker
systemctl start docker
systemctl status docker

It seems, that this is a docker in a lxc container installation.  I wanted a blank linux/lxc installation, without this docker overhead.

**Comment by vigeland** [comment]
I had installed the promoxx script, using the last frigate relase. It ist working well. No docker is running on the lxc instance.

**Reply by GnomiBerlin** [reply]
Thanks for the feedback. As I wrote, I do not want to use Proxmox but will look into the script for the essentials to manually create the lxc container. Just to have no misunderstanding, what exact script (name) do you reference?

**Comment by vigeland** [comment]
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/main/ct/frigate.sh)"

I think it will call this script, after setup the lxc Container.
https://github.com/remz1337/Proxmox/blob/remz/install/frigate-install.sh

**Reply by RussellRego** [reply]
FYI... Does not install in Proxmox 9.1.2.  Message says requires Proxmox Virtual Environment Version 8.1 or later.

**Comment by GnomiBerlin** [comment]
thank you @vigeland for the hint.

to all ;-)
I now spent a lot of time with different tests and with a little adaptation, as I want to bring all running in a pure LXC container on a raspberry pi.
the script `frigate-install.sh` needed to be changed for raspberry usage from 
`export TARGETARCH="amd64"`
to
`export TARGETARCH="arm64"`

**Then it was _basically_ running**  **BUT** could not use the CORAL TPU nor the H.264 HW support from the PI

I switched now back do the **docker usage within the lxc container** and compared the same setup with the docker on the raspberry itself, without lxc. This is perfectly running.

All hints for bringing the TPU into the Proxmox lxc container (adapted from Proxmox to pure linux lxc config) like
```
lxc.cgroup2.devices.allow = c 29:0 rwm
lxc.cgroup2.devices.allow = c 189:* rwm
#lxc.mount.entry: /dev/bus/usb/<BUS_ID> dev/bus/usb/<BUS_ID> none bind,optional,create=dir 0, 0
lxc.mount.entry = /dev/bus/usb/002 /dev/bus/usb/002 none bind,optional,create=dir 0 0
```
were not successful.

The main problem is the Frigate usage from below configuration within the lxc container:
```
ffmpeg:
  hwaccel_args: preset-rpi-64-h264 #Enable Hardware Acceleration

detectors:
  coral:
    type: edgetpu
    device: usb
```

So the main help would be, that someone brought Frigate running in a pure lxc container on a raspberry Pi with TPU usage??

Thanks
